mlibrary/lib.umich.edu
# Replacing Prettier with ESLint.

# Overview
[Prettier](https://www.npmjs.com/package/prettier) was originally set up for the project and only had one rule:
```js
{ 
 "singleQuote": true
}
```

Since the project launched, LIT has adapted [ESLint](https://www.npmjs.com/package/eslint) to become the default linter for JavaScript projects. This pull request completely replaces Prettier with ESLint, and follows similar rules to the [Design System](https://github.com/mlibrary/design-system/blob/main/.eslintrc.json).

## Scripts
### `lint`
`npm run lint` has been added to check all `.js` files that fall under `./src` or `gatsby-*.js`.

### `format`
`npm run format` has been updated to resolve all auto-fixable problems in the files mentioned above using `eslint`.

## `README.md`
The `README.md` listed `prettier` and `yarn` files that no longer exist in the project. It has been cleaned up to help developers run the project.

## Testing
- Install the dependencies (`npm install`).
- Run `npm run lint` to see if any errors are caught.
- Run `npm run format && git status` to see if errors get fixed (do not commit).
